File: Solutions/1417/1417.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 1417.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-8-12 18:20:32
***************************************************************************************************/



char * reformat(char * s){
    char * s1 = (char *)malloc(sizeof(char) * strlen(s));
    char * s2 = (char *)malloc(sizeof(char) * strlen(s));
    int id1=0, id2=0;
    for (int i=0; i<strlen(s); i++){
        if (s[i]<='z' && s[i]>='a') {
            s1[id1++] = s[i];
        } else {
            s2[id2++] = s[i];
        }
    }
    char * ret = (char *)malloc(sizeof(char) * (strlen(s) + 1));
    int id = 0;
    if (id1 - id2 > 1 || id2 - id1 > 1) ret[id] = '\0';
    else if (id1 == id2) {
        for (;id<id1;id++) {
            ret[id*2] = s1[id];
            ret[id*2+1] = s2[id];
        }
        ret[id*2] = '\0';
    } else if (id1 < id2) {
        for (;id<id1;id++) {
            ret[id*2] = s2[id];
            ret[id*2+1] = s1[id];
        }
        ret[id*2] = s2[id];
        ret [id*2+1] = '\0';
    } else {
        for (;id<id2;id++) {
            ret[id*2] = s1[id];
            ret[id*2+1] = s2[id];
        }
        ret[id*2] = s1[id];
        ret[id*2+1] = '\0';
    }
    return ret;
}

File: Solutions/1417/1417.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/8/11 11:13
# File: 1417.py
# Desc: 


class Solution:
    def reformat(self, s: str) -> str:
        s1 = []
        s2 = []
        for i in s:
            if i.isalpha(): s1.append(i)
            else: s2.append(i)
        if abs(len(s1) - len(s2)) > 1: return ''
        if len(s1) > len(s2): s1, s2 = s2, s1
        ret = ''
        for i,j in zip(s1,s2):
            ret += j + i
        if len(s2) > len(s1): ret += s2[-1]
        return ret
